### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by nservidio** [mac-gallagher/Shuffle#26]
Hi there! Thanks for making Shuffle!

I'm having an issue related to https://github.com/mac-gallagher/Shuffle/issues/13. I'm trying to periodically add to content in the `SwipeCardStack`. I'm having to do some fancy stuff with updating the datasource.

Anyways my problem is more specific. If I'm swiping a card and then call `reloadData()` the animations conflict and the cards get stuck in an `isUserInteractionEnabled == false` state. It looks like special care has been taken to disable `isUserInteractionEnabled` for every swipe. Were you experiencing specific issues when not specifically disabling `isUserInteractionEnabled`?

```
func card(didSwipe card: SwipeCard, with direction: SwipeDirection, forced: Bool) {
        delegate?.cardStack?(self, didSwipeCardAt: topCardIndex, with: direction)
        ...
        isUserInteractionEnabled = false
        animator.swipe(self, topCard: card, direction: direction, forced: forced)
    }
```

Please let me know! Thanks,

- Nick

**Comment by CJxD** [comment]
I seem to be getting this too! Initially I was only reloading data when the card stack had been depleted, but now that I am reloading before the end, I get UI deadlocks where the cards can no longer be swiped. Surely this is the same issue.

**Comment by mac-gallagher** [comment]
Hi @nservidio and @CJxD,

Sorry for not responding until now! @nservidio you are correct that care was taken with disabling user interaction - the reason behind this is that if a user swiped a card (manually or programmatically) they can still interact with the card while it is animating, which may cause some unexpected behavior. 

Thanks to you both for bringing up the issues with the `reloadData` call. One possible reason for this is if you are storing strong references to the _cards_ instead of the _models_, so I just want to make sure that both of your `cardForIndexAt` methods look something like this

```swift
func cardStack(_ cardStack: SwipeCardStack, cardForIndexAt index: Int) -> SwipeCard {
    let card = SampleCard()
    card.model = models[index]
    return card
}
```

and not like this

```swift
var cards: [SampleCard] //list of cards

func cardStack(_ cardStack: SwipeCardStack, cardForIndexAt index: Int) -> SwipeCard {
    return cards[index]
}
```

we can further investigate if this is not the case!

**Comment by CJxD** [comment]
Thanks Mac,

I can confirm that I don't store strong card references, but I think the
reason this happens is because I trigger reloadData() from an image load
callback.

Even though I only call the function via the main dispatch queue, there's
the potential of calling it before a card swipe animation is completed,
which messes up the state. This is no more than a guess however.

Hope this helps!

P.S. here's my application of your library:
https://apps.apple.com/bw/app/dankr/id1478419632

**Comment by mac-gallagher** [comment]
@CJxD Great, that narrows it down. 

I was able to reproduce this issue by doing a swipe/undo/shift and then immediately calling `reloadData` before the animation completed. As suspected, the respective completion blocks in `SwipeCardStack` were not being called. As a fix, I added `isUserInteractionEnabled = true` to the `reloadData` call.

Thanks to you both for bringing up the issue! Let me know if this resolves it. The new pod version will be 0.1.3; bumping it now 🙂.

@CJxD looks like a fun app, glad this library could help you out! I'll add your app to the README shortly.

Cheers,
Mac
